web/srv/server.go

```diff
@@ -90,6 +90,7 @@ func NewServer(addr, templateDir, staticDir, uuid, controllerNamespace, webpackD
 	server.router.GET("/replicationcontrollers", handler.handleIndex)
 	server.router.GET("/pods", handler.handleIndex)
 	server.router.GET("/authorities", handler.handleIndex)
+	server.router.GET("/resource", handler.handleIndex)
```

Comment:
How would you feel about making this route `"/resource/:type/:name"`? We should be able to do something like that pretty easily with httprouter, and it would make for somewhat nicer URLs.

Alternatively, if we wanted the resource detail page to be analogous to how we're already handling namespaces, then it might make sense to hardcode all of the resource types:

```
server.router.GET("/pods/:pod", handler.handleIndex)
server.router.GET("/deployments/:deployment", handler.handleIndex)
server.router.GET("/replicationcontrollers/:replicationcontroller", handler.handleIndex)
...
```

That would be more REST-y, too.

Reply:
Yeah! I have been trying to make a decision on that too.
I did alternate between those choices as well. I'd settled on /resource because I realized that we'd need to pass in namespace as well, and it would allow us to handle arbitrary resource types without making any additional changes to the web server, but I think making it REST-y is appealing.

Yeah, it probably would make the most sense to do the hardcoding of the resource type routes so that we can be analogous to how we handle namespaces. For the other resource types, namespace can be a URL parameter, so we can do something like:
`/deployments/voting?namespace=emojivoto`

Reply:
Agreed! I like `/deployments/voting?namespace=emojivoto`, as I feel like that's most consistent with `/namespaces/emojivoto`. Plus it means that the URL paths are valid k8s resources. Or if we wanted to get super REST-y with this, we could do something like `/namespaces/emojivoto/deployments/voting`, but I'm ambivalent about that.

Reply:
Peering just a bit more deeply into this rabbit hole... the Kubernetes API uses scoped resource routes, e.g.:

```
GET /api/v1/namespaces/{namespace}/pods/{name}
GET /apis/apps/v1/namespaces/{namespace}/deployments/{name}
...
```

So it could make sense for us to follow that convention here.

Reply:
I am definitely in favor of having this be more REST-y. I can definitely see this being extended further so that we could also have tap and top-like resources associated for each resource endpoint, for example, if we wanted to get the top results for a deployment, the URL may look something like `/deployments/:deployment/top?top_filters=...` I am not sure if that is worthwhile to do, but it definitely provides for shareable and user-friendly URLs.

Reply:
Yeah! Agreed. Let's follow the k8s API convention for now, and see how we feel.